src/main/scala/cognite/spark/v1/MetricsSource.scala

```diff
@@ -2,22 +2,36 @@ package org.apache.spark.datasource
 
 import cats.Eval
 import com.codahale.metrics._
-import org.apache.spark._
-
 import java.util.concurrent.ConcurrentHashMap
+import org.apache.spark._
 
 class MetricsSource {
   // Add metricNamespace to differentiate with spark system metrics.
   // Keeps track of all the Metric instances that are being published
   val metricsMap = new ConcurrentHashMap[String, Eval[Counter]]
 
-  def getOrCreateCounter(metricNamespace: String, metricName: String): Counter = {
+  def getOrCreateCounter(metricNamespace: String, name: String): Counter = {
+    val ctx = Option(TaskContext.get())
+
+    def getNumberOrEmpty(getter: TaskContext => AnyVal): String =
+      ctx.map(getter).map(_.toString()).getOrElse("")
```

Comment:
```suggestion
    def getNumberOrEmpty(getter: TaskContext => String): String =
      ctx.map(getter).getOrElse("")
```
to be more type-safe, at call-sites just add a quick .toString()

Reply:
Not sure if there's a common numeric type that we could use to make sure getter gets a (non-floating point) number (Long?)

Reply:
`Long` should cover them all, yeah 👍 Especially bc this is a local function